lxd/containers_post.go

```diff
@@ -936,6 +1006,70 @@ func containerFindStoragePool(d *Daemon, project string, req *api.InstancesPost)
 
 	return storagePool, storagePoolProfile, localRootDiskDeviceKey, localRootDiskDevice, nil
 }
+//Check instance limit for project, if it exists
+func instancesLessThanLimit(d *Daemon, project string,  instance api.InstanceType) (bool, error) {
```

Comment:
This should use `s *state.State` rather than `d *Daemon`. The function also ought to be in the `lxd/project` package instead.

I'd expect us to call something like: `project.CheckInstanceLimit(project, type)`
And just have it return an error if we currently are at the limit, or nil if not.

I'd also expect this function to be fairly optimized, so first load the project with the `GetByName` function you're adding to the same package, check if a limit is present and if not return nil immediately so we don't needlessly load all the containers/vms.